docker/file_metrics_collector/file_metrics_collector.py

```diff
@@ -0,0 +1,140 @@
+REGISTRY.unregister(PROCESS_COLLECTOR)
+REGISTRY.unregister(PLATFORM_COLLECTOR)
+
+# remove python-garbage-collectior metrics
+REGISTRY.unregister(
+    REGISTRY._names_to_collectors['python_gc_objects_uncollectable_total'])
+
+WIS2BOX_LOGGING_LOGLEVEL = os.environ.get('WIS2BOX_LOGGING_LOGLEVEL')
+# gotta log to stdout so docker logs sees the python-logs
+logging.basicConfig(stream=sys.stdout)
+# create our own logger using same log-level as wis2box
+logger = logging.getLogger('file_metrics_collector')
+logger.setLevel(WIS2BOX_LOGGING_LOGLEVEL)
+
+INTERRUPT = False
+
+
+def collect_data(root, filematch):
+    """
+    helper-function to collect data on files in given root-path
+
+    :param root: root-path for file search
+    :param filematch: string to be contained in filename
+
+    :returns: array of objects with file-data
+    """
+    from fnmatch import fnmatch
+    data = {}
+    for path, subdirs, files in os.walk(root):
+        for name in files:
+            if fnmatch(name, filematch):
+                # parse wigos-id out of file-name
+                # this will fail when files are formatted differently
+                wsi = ''
+                if 'WIGOS_' in name:
+                    wsi = name.split('WIGOS_')[1].split(
+                        f'_{time.strftime("%Y")}')[0]
+                if wsi not in data:
+                    data[wsi] = {}
+                    data[wsi]['nfiles'] = 1
+                else:
+                    data[wsi]['nfiles'] += 1
+    return data
+
+
+def gather_file_metrics():
+    """
+    gather file metrics by checking content of wis2box data directory
+
+    :returns: `None`
+    """
+    # Gauges to measure files counted in path
+    bufr_out_nfile_total = Gauge(
+        'bufr_out_nrfile_total',
+        'Total bufr4 in /data/public')
+    bufr_out_nfile_wsi_total = Gauge(
+        'bufr_out_nrfile_wsi_total',
+        'Total bufr4 in /data/public, by WIGOS-ID',
+        ["wigos_id"])
+    csv_in_nfile_total = Gauge(
+        'csv_in_nrfile_total',
+        'Total csv in /data/incoming')
+    csv_in_nfile_wsi_total = Gauge(
+        'csv_in_nrfile_wsi_total',
+        'Total csv in /data/incoming, by WIGOS-ID',
+        ["wigos_id"])
+
+    while not INTERRUPT:
+        # count bufr4 in outgoing directory
+        root_public = '/data/wis2box/data/public'
+        bufr_data = collect_data(root_public, filematch='*.bufr4')
+
+        bufr_out_nfiles = 0
+        for wigos_id in bufr_data:
+            bufr_out_nfiles += bufr_data[wigos_id]['nfiles']
+            bufr_out_nfile_wsi_total.labels(wigos_id).set(
+                bufr_data[wigos_id]['nfiles']
+            )
+        logger.info(f"Found {bufr_out_nfiles} .bufr4 in {root_public}")
+        # update gauge for total
+        bufr_out_nfile_total.set(bufr_out_nfiles)
+
+        # count csv in coming directory
+        root_incoming = '/data/wis2box/data/incoming'
```

Comment:
Can some part of this be derived from environment variables?

Reply:
Yes sure I could derive the string to be used for 'root_incoming' from an environment-variable.
Note that I would prefer to remove the 'file_metrics_collector'-process from future wis2box versions all together and either have watchpuppy update the Prometheus-metrics directly or have another container listening to watchpuppy and populate file-metrics. As such I was not too concerned about hard-coding the file-path in this implementation...